git_machete/git_operations.py

```diff
@@ -804,10 +804,53 @@ def is_equivalent_tree_reachable(
         )
 
         result = earlier_tree_hash in intermediate_tree_hashes
-        debug(f"result = {result}")
+        debug(f"intermediate tree hashes result = {result}")
+
+        if result:
+            self.__is_equivalent_tree_reachable_cached[equivalent_to_commit_hash, reachable_from_commit_hash] = result
+            return result
+
+        # Let's try another way, a little more complex but takes into account the possibility
+        # that there were other commits between the fork point of the two branches and the squashed merge.
```

Comment:
Pls note that we strive to keep `git_machete/git_operations.py` as machete-agnostic as possible... and in particular, I wouldn't mention _fork point_ here (which is originally a git term, see `git rebase --fork-point`, but [heavily adapted within git machete](https://medium.com/virtuslab/make-your-way-through-the-git-rebase-jungle-with-git-machete-e2ed4dbacd02#:~:text=Appendix%3A%20fork%20point%20%E2%80%94%20not%20so%20easy%20business%E2%80%A6).

Reply:
Maybe just
```suggestion
        # that there were other commits between the common ancestor of the two branches and the squashed merge.
```
Not sure if that preserves the exact sense 🤔

Reply:
Maybe even better (at least that's what speaks better to me, somehow):
```suggestion
        # that before the `equivalent_to` commit has been squashed onto into the history of `reacheable_from`,
        # more commits appeared on history of `reacheable_from`.
        # Hence, there might NOT be a commit reachable from `reacheable_from` that points to the exact same tree as in `equivalent_to`.
```